README.md
# Ext.Scheduler


changelog.txt
Release history for Ext Scheduler:

2.5.3     2015-02-02
    [SUPPORTED EXT JS VERSIONS]
    - 4.2.1
    - 4.2.2

    [KNOWN ISSUES]

    [FEATURES / ENHANCEMENTS]

    [API CHANGES]

    [BUG FIXES]
    - Fixed #1501: Calendar View internalIds 
    - Fixed #1504: German translation of Export settings dialog
    - Fixed #1517: Calendar View after drag selection visibility is removed
    - Fixed #1760: Add support for a customer header template to the export result
    - Fixed #1761: getCurrentZoomLevelIndex return incorrect results
    - Fixed #1782: Rows are out of sync in vertical scheduler after fitColumns call
    - Fixed #1809: Weekview example should allow to create events with no resource assigned
    - Fixed #1811: Exporter tpl config should be converted into XTemplate
    - Fixed #1816: Export add footerHeight and showFooter properties
    - Fixed #1829: Multi-page export with many locked columns shows wrong columns
    - Fixed #1842: Make AJAX request configurable
    - Fixed #1846: Export dialog "Paper format" combobox has hardcoded formats list


2.5.2     2014-12-22


    [SUPPORTED EXT JS VERSIONS]
    - 4.2.1
    - 4.2.2

    [KNOWN ISSUES]

    [FEATURES / ENHANCEMENTS]
    - Added the proxy element as a parameter to SchedulerView 'dragcreatestart', 'beforedragcreatefinalize', 'dragcreateend' and 'afterdragcreate' event signatures.

    [API CHANGES]

    [BUG FIXES]
    - Fixed #1572: Export plugin needs to work with buffering
    - Fixed #1573: Multi-page export with two rows of pages shows scrollbar and displaced dependencies
    - Fixed #1672: Export date range gives reconfigure error.
    - Fixed #1739: Scheduler always saves a state
    - Fixed #1763: Add timeout property for export plugin
    - Fixed #1773: Events do not load in Tree Scheduler
    - Fixed #1775: Crud manager has to fire loadfail event of AJAX failure 

2.5.1     2014-11-13

    [SUPPORTED EXT JS VERSIONS]
    - 4.2.1
    - 4.2.2

    [KNOWN ISSUES]

    [FEATURES / ENHANCEMENTS]

    [API CHANGES]

    [BUG FIXES]
    - Fixed #1469: Scheduler timeline doesn't occupy scheduler view entirely on big resolutions
    - Fixed #1691: Crash during event resize

2.5.0     2014-11-07
    This release includes two new features: CrudManager for easier loading and saving of schedule data (see updated "print" sample) and
    a calendar week view (see new "weekview" sample). Additionally a lot of bugs were fixed, listed below.

    [SUPPORTED EXT JS VERSIONS]
    - 4.2.1
    - 4.2.2

    [KNOWN ISSUES]

    [FEATURES / ENHANCEMENTS]
    - Introduced new config 'mode' and methods getMode/setMode. In further releases 'orientation' is going to be deprecated
    in favor of 'mode'

    [API CHANGES]
    - Signature of Sch.model.view.TimeAxis#getPositionFromDate method is enhanced: new optional argument isEnd is added.
    Required only for calendar view
    - Added a new event 'eventrepaint' to the SchedulerView class which fires after an event has been repainted.
    - Added new "setViewPreset" / "getViewPreset" methods to the Sch.mixin.TimelinePanel class.
    - Added a new "setRowHeight" method to the Sch.mixin.SchedulerPanel class.

    [BUG FIXES]
    - Fixed #1558: Normal grid changes time span on splitter move when infinite scroll enabled
    - Fixed #1571: Print plugin does not work correctly when inside a viewport.
    - Fixed #1577: Scheduler / Ext 4.2.3.1477: Event resize in TimeLine not working correct
    - Fixed #1606: Model.set override breaks Ext JS API
    - Fixed #1619: IE Patch TreeViewDragDrop error on enableDrag = false
    - Fixed #1624: Tick width isn't reduced when scheduler width is
    - Fixed #1638: Crash when not specifying start date / end date
    - Fixed #1642: DST issue in Brazilian timezone
    - Fixed #1648: EventEditor should be repositioned after event update
    - Fixed #1661: Mixed use of line endings, potential source of headaches
    - Fixed #1666: Event tooltip cannot be anchored

2.2.25    2014-09-10

    [SUPPORTED EXT JS VERSIONS]
    - 4.2.1
    - 4.2.2

    [KNOWN ISSUES]

    [FEATURES / ENHANCEMENTS]

    [API CHANGES]

    [BUG FIXES]
    - Fixed #1496: Circular reference in requirements chain - Localizable.js
    - Fixed #1521: zoomToSpan works incorrect for Quarter resolution
    - Fixed #1535: Event rendering bug with grouping feature
    - Fixed #1537: Export plugin cuts the row
    - Fixed #1539: Export example doesn't support output path change
    - Fixed #1544: Gantt+Scheduler combo produces JS error upon Dynamicaly Assigned task effort second time updating
    - Fixed #1550: Filter on EventStore should update event cache
    - Fixed #1555: Drag&drop of multiple events works incorrectly at second time
    - Fixed #1556: Multipage export performance and hidden columns

2.2.24    2014-08-01

    [SUPPORTED EXT JS VERSIONS]
    - 4.2.1
    - 4.2.2

    [KNOWN ISSUES]

    [FEATURES / ENHANCEMENTS]

    [API CHANGES]

    [BUG FIXES]
    - Fixed #1492: ResourceZones performance improvements
    - FIXED #1496: Circular reference in requirements chain - Localizable.js
    - Fixed #1500: Missing requirements
    - Fixed #1515: Multi-page export shows displaced today-line
    - Fixed #1524: Fit columns not working properly in vertical mode
    - Fixed #1506: Export plugin doesn't support provided dateRangeFormat
    - Fixed #1530: Milestone events overlap in scheduler
    - FIXED #1539: Export example doesn't support output path change

2.2.23    2014-06-26

    [SUPPORTED EXT JS VERSIONS]
    - 4.2.1
    - 4.2.2

    [KNOWN ISSUES]

    [FEATURES / ENHANCEMENTS]

    [API CHANGES]

    [BUG FIXES]
    - Fixed #1478: Circular reference in Sch.locale.En

2.2.22    2014-05-30

    [SUPPORTED EXT JS VERSIONS]
    - 4.2.1
    - 4.2.2

    [KNOWN ISSUES]

    [FEATURES / ENHANCEMENTS]

    - Fixed #1374: Add tooltipConfig for dd, create and resize events

    [API CHANGES]

    [BUG FIXES]
    - Fixed #1433: Drag create and resize tooltips get cut off

vertical.js
Ext.ns('App');
Ext.Loader.setConfig({ enabled : true, disableCaching : true });
Ext.Loader.setPath('Sch', '../../js/Sch');

Ext.require([
    'Sch.panel.SchedulerGrid',
    'Sch.plugin.Zones'
]);


Ext.onReady(function () {
    App.SchedulerDemo.init();
});

App.SchedulerDemo = {

    // Initialize application
    init : function () {
        Ext.define('Event', {
            extend : 'Sch.model.Event',
            fields : [
                {name : 'Title'},
                {name : 'Type'}
            ]
        });

        var zoneStore = Ext.create('Ext.data.JsonStore', {
            model : 'Sch.model.Range',
            data  : [
                {
                    // Nice 2 hour lunch
                    StartDate : new Date(2011, 11, 9, 12),
                    EndDate   : new Date(2011, 11, 9, 14),
                    Cls       : 'lunch-style'
                }
            ]
        });

        var sched = App.sched = Ext.create("Sch.panel.SchedulerGrid", {
            height                  : ExampleDefaults.height,
            width                   : ExampleDefaults.width,
            rowHeight               : 40,
            eventBarTextField       : 'Title',
            viewPreset              : 'hourAndDay',
            startDate               : new Date(2011, 11, 9, 7),
            endDate                 : new Date(2011, 11, 9, 17),
            orientation             : 'vertical',
            constrainDragToResource : false,
            snapToIncrement         : true,
            //constrainDragToResource : true,
            eventResizeHandles      : 'end',
            resourceColumnWidth     : 140,

            // For horizontal mode
            columns                 : [
                {
                    text      : 'Name',
                    width     : 200,
                    sortable  : true,
                    dataIndex : 'Name'
                }
            ],

            viewConfig : {
                // Experimental for CSS3 enabled browsers only
                altColCls       : '',
                stripeRows      : false,
                eventAnimations : true
            },

            eventBodyTemplate : new Ext.XTemplate(
                '<span class="time">{[Ext.Date.format(values.StartDate, "G:i")]}</span> {Title}'
            ),

            eventRenderer : function (event, resource, data) {
                data.cls = resource.data.Name;
                return event.data;
            },

            lockedViewConfig : {
                getRowClass : function (resource) {
                    return resource.data.Name;
                }
            },

            timeAxisColumnCfg : {
                text : 'Time of day'
            },

            // Store holding all the resources
            resourceStore     : Ext.create("Sch.data.ResourceStore", {
                model : 'Sch.model.Resource',
                data  : [
                    {Id : 'toJesrel', Name : 'Jesrel'},
                    {Id : 'toKevin', Name : 'Kevin'},
                    {Id : 'toWilfred', Name : 'Wilfred'},
                    {Id : 'toRichie', Name : 'Richie'},
                    {Id : 'toK1', Name : 'K1'},
                    {Id : 'toRany', Name : 'Rany'},
                    {Id : 'toLeysa', Name : 'Leysa'},
                    {Id : 'toRonquillo', Name : 'Ronquillo'},
                    {Id : 'toAgang', Name : 'Agang'}
                ]
            }),

            // Store holding all the events
            eventStore        : Ext.create("Sch.data.EventStore", {
                model : 'Event',
                data  : [
                    {ResourceId : 'toJesrel', Type : 'Call', Title : 'Assignment 1', StartDate : "2011-12-09 10:00", EndDate : "2011-12-09 11:00"},
                    {ResourceId : 'toWilfred', Type : 'Call', Title : 'Customer call', StartDate : "2011-12-09 14:00", EndDate : "2011-12-09 16:00"},
                    {ResourceId : 'toRany', Type : 'Meeting', Title : 'Assignment 2', StartDate : "2011-12-09 10:00", EndDate : "2011-12-09 12:00"}
                ]
            }),

//            plugins : [
//                this.zonePlugin = Ext.create("Sch.plugin.Zones", {
//                    store : zoneStore
//                })
//            ],

            onEventCreated : function (newEventRecord) {
                // Overridden to provide some defaults before adding it to the store
                newEventRecord.set({
                    Title : "Hey, let's meet",
                    Type  : 'Meeting'
                });
            },


            listeners : {
                columnwidthchange : function (timeAxisViewModel, width) {
                    sched && sched.getDockedItems('toolbar')[ 0 ].child('slider').setValue(width)
                }
            },

            tbar : [
                {
                    text         : 'Vertical view',
                    pressed      : true,
                    iconCls      : 'icon-vertical',
                    enableToggle : true,
                    toggleGroup  : 'orientation',
                    handler      : function () {
                        sched.setOrientation('vertical');
                    }
                },
                {
                    text         : 'Horizontal view',
                    enableToggle : true,
                    iconCls      : 'icon-horizontal',
                    toggleGroup  : 'orientation',
                    handler      : function () {
                        sched.setOrientation('horizontal');
                    }
                },
                '->',
                {
                    text    : 'Fit columns',
                    handler : function () {
                        Ext.getCmp('col-slider').suspendEvents();
                        sched.getSchedulingView().fitColumns();
                        Ext.getCmp('col-slider').resumeEvents();
                    }
                },
                'Column Width:',
                {
                    id    : 'col-slider',
                    xtype : 'slider',
                    width : 100,

                    value     : 160,
                    increment : 10,
                    minValue  : 30,
                    maxValue  : 200,

                    listeners : {
                        change : function (slider, value) {
                            sched.getSchedulingView().setColumnWidth(value);
                        }
                    }
                },
                ' ',
                'Row Height:',
                {
                    xtype : 'slider',
                    width : 100,

                    value     : 60,
                    increment : 10,
                    minValue  : 30,
                    maxValue  : 150,

                    listeners : {
                        change : function (sli, v) {
                            var schedulingView = sched.getSchedulingView();
                            schedulingView.setRowHeight(v);
                        }
                    }
                }
            ]
        });

        sched.render(Ext.getBody());
    }
};
